src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1461,20 +1472,35 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
      *
      * @param pos       the position to transfer it from.
      * @param predicate the predicate to evaluate.
+     * @param stackInfo the actual stack of the predicate to get the needed value
      * @return true if succesful.
      */
-    private boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate)
+
+    // modified to reflect the actual stack amount to transfer.  WIP
+    private boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate, @NotNull final BuildingBuilderResource stackInfo)
     {
         final TileEntity entity = world.getTileEntity(pos);
         if (entity == null)
         {
             return false;
         }
 
+        // Filter out if Size to pick is bigger than stacksize
+
+        int insertAmount =0;
+        if(stackInfo.getAmount() > stackInfo.getItemStack().getMaxStackSize())
+        {
+            insertAmount = stackInfo.getItemStack().getMaxStackSize();
+        }else{
```

Comment:
can you format this properly?
In the repo there is a formatting file for intellij you can setup to properly format this too.

Reply:
formatting done